app/src/main/java/de/westnordost/streetcomplete/data/osm/osmquest/OsmFilterQuestType.kt

```diff
@@ -16,11 +15,11 @@ abstract class OsmFilterQuestType<T> : OsmElementQuestType<T> {
         /* this is a considerate performance improvement over just iterating over the whole MapData
         *  because for quests that only filter for one (or two) element types, any filter checks
         *  are completely avoided */
-        val iterable = MultiIterable<Element>()
-        if (filter.includesElementType(Element.Type.NODE)) iterable.add(mapData.nodes)
-        if (filter.includesElementType(Element.Type.WAY)) iterable.add(mapData.ways)
-        if (filter.includesElementType(Element.Type.RELATION)) iterable.add(mapData.relations)
-        return iterable.filter { element -> filter.matches(element) }
+        var iterable = sequenceOf<Element>()
+        if (filter.includesElementType(Element.Type.NODE)) iterable += mapData.nodes
+        if (filter.includesElementType(Element.Type.WAY)) iterable += mapData.ways
+        if (filter.includesElementType(Element.Type.RELATION)) iterable += mapData.relations
+        return iterable.filter { element -> filter.matches(element) }.toList()
```

Comment:
No matter if it is a sequence or an iterable before. If you `toList` it, you copy everything into an array list. This is a huge performance loss, as this means that for most quest types, a large portion of the downloaded data is copied.

Reply:
Okay, I'll have a look what can be done instead here.

Reply:
`toList` can be avoided like this: https://stackoverflow.com/a/51464098/451391